File: .notes/namespaces/Julia/iife-cell.jl


# ── demo ────────────────────────────────────────
cell = 
( () -> begin 
	
	a = 1 ;
	b = a + 1 ;
	c = (n) -> a + b * n ;

( a = a, c = c ) end )()

# ── effect ──────────────────────────────────────
cell.a #> 1
cell.c(1) #> 3
cell.b #!> ERROR: type NamedTuple has no field b

File: .notes/namespaces/Rust/iife-cell.rs
fn main() 
{
	/* ── demo ──────────────────────────────────────── */
	let cell = 
	(|| {
		
		let add = |x: i32, y: i32| -> i32 { x + y };
		let sss = |x: &str, y: &str| -> String { x.to_owned() + y };
		let a = 1;
		let b = "booooo";
		let _c = "some thing won't export ...";
		
		struct Yrd <'s>
		{
			add: fn(i32, i32) -> i32, 
			sss: fn(&str, &str) -> String, 
			a: i32, 
			b: &'s str, 
		}
		
		Yrd 
		{
			add: add, 
			sss: sss, 
			a: a, 
			b: b, 
		}
	}) ();
	
	/* ── effect ────────────────────────────────────── */
	println!("{}", (cell.add)(cell.a, 2));     //> 3
	println!("{}", (cell.sss)(cell.b, "yyy")); //> boooooyyy
}
